File: Creating a MVC 3 Application with Razor and Unobtrusive JavaScript/[VB]-Creating a MVC 3 Application with Razor and Unobtrusive JavaScript/VB/MvcRazorVB/Controllers/HomeController.vb
Public Class HomeController
    Inherits System.Web.Mvc.Controller

    ' The __usrs class is replacement for a real data access strategy.
    Private Shared _usrs As New Users()

    Function Index() As ActionResult
    	Return View(_usrs._usrList)
    End Function

    Public Function Details(ByVal id As String) As ViewResult
        Return View(_usrs.GetUser(id))
    End Function

    Public Function Edit(ByVal id As String) As ViewResult
        Return View(_usrs.GetUser(id))
    End Function

    <HttpPost()>
    Public Function Edit(ByVal um As UserModel) As ViewResult

        If Not TryUpdateModel(um) Then
            ViewBag.updateError = "Update Failure"
            Return View(um)
        End If

        ' ToDo: add persistent to DB.
        _usrs.Update(um)
        Return View("Details", um)
    End Function

    Public Function About() As ActionResult
        Return View()
    End Function

    Public Function Create() As ViewResult
        Return View(New UserModel())
    End Function

    <HttpPost()>
    Public Function Create(ByVal um As UserModel) As ViewResult

        If Not TryUpdateModel(um) Then
            ViewBag.updateError = "Create Failure"
            Return View(um)
        End If

        ' ToDo: add persistent to DB.
        _usrs.Create(um)
        Return View("Details", um)
    End Function

    Public Function Delete(ByVal id As String) As ViewResult
        Return View(_usrs.GetUser(id))
    End Function

    <HttpPost()>
    Public Function Delete(ByVal id As String, ByVal collection As FormCollection) As RedirectToRouteResult
        _usrs.Remove(id)
        Return RedirectToAction("Index")
    End Function

End Class

File: Creating a MVC 3 Application with Razor and Unobtrusive JavaScript/[C#]-Creating a MVC 3 Application with Razor and Unobtrusive JavaScript/C#/Mvc3Razor/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using Mvc3Razor.Models;

namespace Mvc3Razor.Controllers {
    public class HomeController : Controller {

        // The __usrs class is replacement for a real data access strategy.
        private static Users _usrs = new Users();

        public ActionResult Index() {
            return View(_usrs._usrList);
        }

        public ViewResult Details(string id) {
            return View(_usrs.GetUser(id));
        }

        public ViewResult Edit(string id) {
            return View(_usrs.GetUser(id));
        }

        [HttpPost]
        public ViewResult Edit(UserModel um) {

            if (!TryUpdateModel(um)) {
                ViewBag.updateError = "Update Failure";
                return View(um);
            }

            // ToDo: add persistent to DB.
            _usrs.Update(um);
            return View("Details", um);
        }

        public ActionResult About() {
            return View();
        }

        public ViewResult Create() {
            return View(new UserModel());
        }

        [HttpPost]
        public ViewResult Create(UserModel um) {

            if (!TryUpdateModel(um)) {
                ViewBag.updateError = "Create Failure";
                return View(um);
            }

            // ToDo: add persistent to DB.
            _usrs.Create(um);
            return View("Details", um);
        }

        public ViewResult Delete(string id) {
            return View(_usrs.GetUser(id));
        }

        [HttpPost]
        public RedirectToRouteResult Delete(string id, FormCollection collection) {
            _usrs.Remove(id);
            return RedirectToAction("Index");
        }


    }
}
